src/System.Diagnostics.Process/src/System/Diagnostics/ProcessManager.Windows.cs

```diff
@@ -50,14 +50,28 @@ public static bool IsProcessRunning(int processId, string machineName)
         /// <returns>The ProcessInfo for the process if it could be found; otherwise, null.</returns>
         public static ProcessInfo GetProcessInfo(int processId, string machineName)
         {
-            ProcessInfo[] processInfos = ProcessManager.GetProcessInfos(machineName);
-            foreach (ProcessInfo processInfo in processInfos)
+            if (IsRemoteMachine(machineName))
             {
-                if (processInfo.ProcessId == processId)
+                // remote case: we take the hit of looping through all results
+                ProcessInfo[] processInfos = NtProcessManager.GetProcessInfos(machineName, isRemoteMachine: true);
+                foreach (ProcessInfo processInfo in processInfos)
                 {
-                    return processInfo;
+                    if (processInfo.ProcessId == processId)
+                    {
+                        return processInfo;
+                    }
+                }
+            }
+            else
+            {
+                // local case: we attempt to get the matching (by pid) process only
+                ProcessInfo[] processInfos = NtProcessInfoHelper.GetProcessInfos(processId);
```

Comment:
The reason why this is failing in UAP is because it is not using the one you declared in ProcessManager.Win32.cs, but instead it is trying to use the one that lives later on this file which is private (on line 695 of this file). You will need to fix that befor merging to get a green uap build. This will fix the three remaining errors you have on your CI build.

Reply:
Thanks, @joperezr, I cannot, however, see why that would be the case...

I'm calling a method here that takes one parameter. The one one line 695 in the same file takes two required parameters so the compiler should not complain about visibility or anything but instead call the one in the other partial class in ProcessManager.Win32.cs... Also, when I CTRL-click on the call, VS also takes me to the correct implementation.

I'm thinking that something else must be wrong in my setup... Will keep digging. Next test will be without that optional parameter in `GetProcessInfos(int processId = AllProcessIds)` in ProcessManager.Win32.cs. Maybe that causes confusion in conjunction with the partial types or something...

Reply:
Riiiight. I just re-read your comment above about the .Win32.cs file:

> UAP and UAPAOT Configurations don't build this file, which is why CI is failing. You can try moving this const to the other file.

So that's got to be my problem. Mind you, I did run an `msbuild /t:Rebuild /p:TargetGroup=uap` locally and I saw those .Win32.cs files in the log so I assumed they were part of the build for some reason. Oh well, I'll get there eventually, bear with me.

Reply:
No problem at all, looks like your build is working now, UWP build hasn't finished but that is due to an infrastructure problem and it is just waiting on running your tests. @MattGal is there an ETA on when will that problem get fixed?